app/services/embeddable_cache_refresh_service.rb

```diff
@@ -0,0 +1,111 @@
+# frozen_string_literal: true
+
+class EmbeddableCacheRefreshService
+  EMBEDDABLE_CACHE_REFRESH_URL = "https://api.us.embeddable.com/api/v1/caching/refresh-contexts".freeze
+
+  attr_accessor :params
+
+  def initialize(params)
+    self.params = params
+  end
+
+  def refresh_contexts
+    request_cache_refresh(params)
+  rescue HTTParty::ResponseError => e
+    {
+      status: :unprocessable_entity,
+      errors: { base: [ "Failed to refresh cache contexts from Embeddable: #{e.message}" ] },
+    }
+  rescue JSON::ParserError => e
+    {
+      status: :unprocessable_entity,
+      errors: { base: [ "Failed to parse response from Embeddable: #{e.message}" ] },
+    }
+  rescue StandardError => e
+    {
+      status: :unprocessable_entity,
+      errors: { base: [ "Unexpected error: #{e.message}" ] },
+    }
+  end
+
+private
+
+  def request_cache_refresh(params)
+    headers = embeddable_headers
+    payload = build_refresh_payload(params)
+
+    response = HTTParty.post(
+      EMBEDDABLE_CACHE_REFRESH_URL,
+      headers: headers,
+      body: payload.to_json,
+      timeout: 30
+    )
+
+    if response.body.blank?
+      { success: true, status: response.code }
+    else
+      JSON.parse(response.body)
+    end
+  end
+
+  def embeddable_headers
+    {
+      "Content-Type"  => "application/json",
+      "Accept"        => "application/json",
+      "Authorization" => "Bearer #{embeddable_api_key}",
+    }
+  end
+
+  def embeddable_api_key
+    ENV.fetch("EMBEDDABLE_API_KEY") do
+      raise StandardError, "EMBEDDABLE_API_KEY environment variable is required"
```

Comment:
I dont think this is necessary.
ENV.fetch will raise KeyError

So instead of having
StandardError: EMBEDDABLE_API_KEY environment variable is required

We will have:
KeyError: key not found: "EMBEDDABLE_API_KEY"

Does this work for you?

Reply:
I think the same